src/acc/opencl/smm/tune_multiply.py

```diff
@@ -128,14 +128,14 @@ def run(self, desired_result, input, limit):
             return Result(time=float("inf"), accuracy=0.0, size=100.0)
 
     def merge_into_csv(self, filenames):
-        """merge all JSONs into a single CSV file"""
+        """merge all JSONs into a single CSV-file"""
         if self.args.csvfile:
             merged = dict()
             for ifilename in filenames:
                 with open(ifilename, "r") as ifile:
-                    data = json.load(ifile)
                     try:
-                        key = (data["TYPEID"], data["M"], data["N"], data["K"])
+                        data = json.load(ifile)
```

Comment:
since you don't need the file anymore after `json.load` I usually do this to reduce indentation levels and make it a bit clearer (Python is not as strictly scoped as C/C++)

```python
with open(ifilename, "r") as ifile:
    data = json.load(ifile)
try:
    key = ...
```

Reply:
Hmm, let me change the structure a bit more (good point). I moved `data = json.load(ifile)` underneath of `try` to capture read-errors.